test/gtest/ucp/test_ucp_tag.cc

```diff
@@ -601,32 +607,35 @@ class test_ucp_tag_nbx : public test_ucp_tag {
             return;
         }
 
-        int rcache_prev;
-        rcache_prev = UCS_STATS_GET_COUNTER(sender().ucph()->rcache->stats,
-                                            UCS_RCACHE_GETS);
+        const ucp_ep_config_t *config = ucp_ep_config(sender().ep());
+        if (send_dt.buf_length() >= config->tag.rndv.am_thresh.local &&
+            is_iov() && enable_proto()) {
+            /* For rndv proto v2 + IOV datatype, we send an RTR request,
+             * which uses rcache, so counters test is not relevant. */
```

Comment:
would the test fail in such case?

Reply:
not sure I understand: you mean will the counters verification fail? or the rest of the test?

Reply:
why we have to skip?

Reply:
because rcache counters are increased each call to tag_send, so it fails the ASSERT in line 623

Reply:
why are they increased?

Reply:
AFAIU, internal rndv requests (RTR/RTS) do not use user memh, so they call rcache

Reply:
I'd expect rndv/am/zcopy be used on sender side. so it should also increase rcache stats.
BTW we should check rcache stats in receiver side as well - they should increase if we use rndv with contig datatype.

Reply:
but those are exactly the cases I don't want to check (I'm only interested in cases where rcache counters are not increased)

Reply:
problem solved: was a bug in detecting IOV user memh